### PR TITLE
watch module: changed internal buffer to query for multi-client support

### PR DESCRIPTION
**Pull request by paradix**

### CODE REVIEW

**Comment by DarthGandalf** [comment]
Can you explain? The change looks like removing all the features.

**Comment by paradix** [comment]
Well, I was trying to make the watch module work all together with the clientbuffer module.
So that you have the possibility to play the matches from watch on each of the used clients separately. 
With the internal CBuffer used in CWatcherMod it was not working, so I replaced it with a CQuery for each watched pattern. 
If the client isn't offline it saves a query which is than played back on login. Seemed to work for me when testing with two different clients.

File: modules/watch.cpp [hunk 1]
```diff
@@ -377,10 +357,14 @@ class CWatcherMod : public CModule {
                                       "!watch@znc.in PRIVMSG " +
                                       pNetwork->GetCurNick() + " :" + sMessage);
                 } else {
-                    m_Buffer.AddLine(
+                    CQuery* pQuery = pNetwork->AddQuery(WatchEntry.GetTarget());
+                    if (pQuery) {
+                        
+                        pQuery->AddBuffer(
                         ":" + _NAMEDFMT(WatchEntry.GetTarget()) +
```
Comment (DarthGandalf) [inline modules/watch.cpp]: clang-format would make this nicer

File: modules/watch.cpp [hunk 2]
```diff
@@ -174,11 +174,61 @@ class CWatchEntry {
 class CWatcherMod : public CModule {
   public:
     MODCONSTRUCTOR(CWatcherMod) {
-        Load();
+        AddHelpCommand();
+        AddCommand("Add", static_cast<CModCommand::ModCmdFunc>(&CWatcherMod::Watch), "<HostMask> [Target] [Pattern]", "Used to add an entry to watch for.");
```
[the remark this answers is not in the document]
Comment (paradix) [reply modules/watch.cpp]: Changed with commit 7d5562a9bdcc810244443a2982135577f0fc8848

File: modules/watch.cpp [hunk 3]
```diff
@@ -174,11 +174,61 @@ class CWatchEntry {
 class CWatcherMod : public CModule {
   public:
     MODCONSTRUCTOR(CWatcherMod) {
-        Load();
+        AddHelpCommand();
+        AddCommand("Add", static_cast<CModCommand::ModCmdFunc>(&CWatcherMod::Watch), "<HostMask> [Target] [Pattern]", "Used to add an entry to watch for.");        
+        AddCommand("List", static_cast<CModCommand::ModCmdFunc>(&CWatcherMod::List), "", "List all entries being watched.");
```
[the remark this answers is not in the document]
Comment (paradix) [reply modules/watch.cpp]: Changed with commit 7d5562a9bdcc810244443a2982135577f0fc8848

File: modules/watch.cpp [hunk 4]
```diff
@@ -506,7 +517,7 @@ class CWatcherMod : public CModule {
         }
     }
 
-    void Dump() {
+    void Dump(const CString& line) {
```
[the remark this answers is not in the document]
Comment (paradix) [reply modules/watch.cpp]: Changed with commit 7d5562a9bdcc810244443a2982135577f0fc8848

File: modules/watch.cpp (hunk 1 above)
Comment (paradix) [reply modules/watch.cpp]: I ran clang format on this part and got no changes compared to what is commited. Or maybe I do not understand you comment :)

Comment (DarthGandalf) [reply modules/watch.cpp]: Strange, it should have reindented it so that parameters are to the right of the function call.

Comment (paradix) [reply modules/watch.cpp]: you mean like this?

Comment (DarthGandalf) [reply modules/watch.cpp]: No, but it's better than no offset at all. Are you sure you've run it?

Comment (paradix) [reply modules/watch.cpp]: yes, it's running on my znc right now.

Comment (DarthGandalf) [reply modules/watch.cpp]: Sorry, by "it" I meant clang-format. I doubt you're running clang-format on your ZNC right now :)

Comment (paradix) [reply modules/watch.cpp]: haha 🤦‍♂

**Pull request merged**